File: diploma-web/src/main/java/org/arhor/diploma/DiplomaApp.java
package org.arhor.diploma;

import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;

@SpringBootApplication
public class DiplomaApp {

  public static void main(String[] args) {
    SpringApplication.run(DiplomaApp.class, args);
  }

  @Bean
  public CommandLineRunner run() {
    return args -> {};
  }
}

File: diploma-server/src/main/kotlin/org/arhor/diploma/DiplomaApp.kt
package org.arhor.diploma

import org.arhor.diploma.commons.StartupTask
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

fun main(args: Array<String>) {
    runApplication<DiplomaApp>(*args)
}

@SpringBootApplication
class DiplomaApp(private val startupTasks: List<StartupTask>) : CommandLineRunner {

    override fun run(vararg args: String) {
        startupTasks.sorted().forEach(StartupTask::execute)
    }
}
